identity/v3userpass_test.go

```diff
@@ -0,0 +1,88 @@
+package identity
+
+import (
+	gc "gopkg.in/check.v1"
+
+	"gopkg.in/goose.v1/testing/httpsuite"
+	"gopkg.in/goose.v1/testservices/identityservice"
+)
+
+type V3UserPassTestSuite struct {
+	httpsuite.HTTPSuite
+}
+
+var _ = gc.Suite(&V3UserPassTestSuite{})
+
+func (s *V3UserPassTestSuite) TestAuthAgainstServer(c *gc.C) {
+	service := identityservice.NewV3UserPass()
+	service.SetupHTTP(s.Mux)
+	userInfo := service.AddUser("joe-user", "secrets", "tenant")
+	var l Authenticator = &V3UserPass{}
+	creds := Credentials{
+		User:    "joe-user",
+		URL:     s.Server.URL + "/v3/auth/tokens",
+		Secrets: "secrets",
+	}
+	auth, err := l.Auth(&creds)
+	c.Assert(err, gc.IsNil)
+	c.Assert(auth.Token, gc.Equals, userInfo.Token)
+}
+
+func (s *V3UserPassTestSuite) TestAuthToAProject(c *gc.C) {
+	service := identityservice.NewV3UserPass()
+	service.SetupHTTP(s.Mux)
+	userInfo := service.AddUser("joe-user", "secrets", "tenant")
+	var l Authenticator = &V3UserPass{}
+	creds := Credentials{
+		User:       "joe-user",
+		URL:        s.Server.URL + "/v3/auth/tokens",
+		Secrets:    "secrets",
+		TenantName: "tenant",
+	}
+	auth, err := l.Auth(&creds)
+	c.Assert(err, gc.IsNil)
+	c.Assert(auth.Token, gc.Equals, userInfo.Token)
+	c.Assert(auth.TenantId, gc.Equals, userInfo.TenantId)
+}
+
+func (s *V3UserPassTestSuite) TestAuthWithCatalog(c *gc.C) {
```

Comment:
Please, add another test where catalog is missing, but token is present (see earlier comment).